components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/util/ResponseTypeHandlerUtil.java

```diff
@@ -0,0 +1,617 @@
+                        } else {
+                            log.debug("Infinite lifetime Access Token found in cache");
+                        }
+                    }
+
+                    if (cacheEnabled) {
+                        OAuthCache.getInstance().addToCache(cacheKey, existingAccessTokenDO);
+                        if (log.isDebugEnabled()) {
+                            log.debug("Access Token was added to cache for cache key : "
+                                    + cacheKey.getCacheKeyString());
+                        }
+                    }
+                    return existingAccessTokenDO;
+
+                } else {
+
+                    if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable
+                            (IdentityConstants.IdentityTokens.ACCESS_TOKEN)) {
+                        log.debug("Access Token is " + existingAccessTokenDO.getTokenState());
+                    }
+                    String tokenState = existingAccessTokenDO.getTokenState();
+
+                    if (OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE.equals(tokenState)) {
+
+                        // Token is expired. If refresh token is still valid, use it.
+                        if (refreshTokenExpiryTime > 0 || refreshTokenExpiryTime < 0) {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Access token has expired, But refresh token is still valid. User existing " +
+                                        "refresh token.");
+                            }
+                            refreshToken = existingAccessTokenDO.getRefreshToken();
+                            refreshTokenIssuedTime = existingAccessTokenDO.getRefreshTokenIssuedTime();
+                            refreshTokenValidityPeriodInMillis =
+                                    existingAccessTokenDO.getRefreshTokenValidityPeriodInMillis();
+                        }
+
+                        if (log.isDebugEnabled()) {
+                            log.debug("Marked Access Token as expired");
+                        }
+                    } else {
+
+                        //Token is revoked or inactive
+                        if (log.isDebugEnabled()) {
+                            log.debug("Access Token is " + existingAccessTokenDO.getTokenState());
+                        }
+                    }
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("No access token found in database for Client ID : " + consumerKey +
+                            ", User ID : " + authorizedUser + " and Scope : " + scope +
+                            ". Therefore issuing new access token");
+                }
+            }
+
+            Timestamp timestamp = new Timestamp(new Date().getTime());
+
+            // if reusing existing refresh token, use its original issued time
```

Comment:
Start comment with capital letter and end the comment with a full stop. Please check other places as well.